### ISSUE TITLE
Add Angular 16 support

### ISSUE THREAD

**Issue by Ajax-ua** [neroniaky/angular2-token#572]
ModuleWithProviders without a generic type is deprecated in Angular 16. So the package doesn't work after migration from Angular 15 to v16

**Comment by rfroetscher** [comment]
@neroniaky Do y'all have plans to support angular 16?

**Comment by vmk-dev** [comment]
what about Angular 16/17 support? 
Is this library dead already?

**Comment by fabricioflores** [comment]
@Ajax-ua did you find a solution? or another library to manage devise token auth?

**Comment by Ajax-ua** [comment]
@fabricioflores No, seems like there is no alternative lib for that. We decided to implement our own solution

**Comment by raysuelzer** [comment]
So, it looks like in the library the underlying code is using things like CanActivate which was removed in version 16. The issue with the generic type argument is strange, because it does use a generic type argument in the code, but the generated .d.ts that is shipped does not include the generic. This is an easy fix as you simply need to override that to be ModuleWithProviders<any>. But, it still breaks in Angular 16, I suspect due to the fact it is using a bunch of deprecated features. 

I'm hoping that someone forks this and makes a compatible version.

**Comment by zandremoku** [comment]
I did support until version 21 forking here: https://github.com/zandremoku/angular-token
